File: src/com/eagle/programmar/BNF/BNF_Rule.java
// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Jul 15, 2015

package com.eagle.programmar.BNF;

import com.eagle.programmar.BNF.Symbols.BNF_Rule_Definition;
import com.eagle.programmar.BNF.Terminals.BNF_Punctuation;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationSemicolon;

public class BNF_Rule extends TokenSequence
{
	public @S(10) BNF_Rule_Definition definition;
	public @S(20) BNF_Punctuation coloncolonEquals = new BNF_Punctuation("::=");
	public @S(30) BNF_Expression expression;
	public @S(40) PunctuationSemicolon semicolon;
}

File: csrc/com/eagle/programmar/BNF/BNF_Rule.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization, 2010-date
// Original author: Steven A. O'Hara, Jul 15, 2015

namespace com.eagle.programmar.BNF
{
	using BNF_Rule_Definition = com.eagle.programmar.BNF.Symbols.BNF_Rule_Definition;
	using BNF_Punctuation = com.eagle.programmar.BNF.Terminals.BNF_Punctuation;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationSemicolon = com.eagle.tokens.punctuation.PunctuationSemicolon;

	public class BNF_Rule : TokenSequence
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.BNF.Symbols.BNF_Rule_Definition definition;
		public BNF_Rule_Definition definition;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.BNF.Terminals.BNF_Punctuation coloncolonEquals = new com.eagle.programmar.BNF.Terminals.BNF_Punctuation("::=");
		public BNF_Punctuation coloncolonEquals = new BNF_Punctuation("::=");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) BNF_Expression expression;
		public BNF_Expression expression;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.tokens.punctuation.PunctuationSemicolon semicolon;
		public PunctuationSemicolon semicolon;
	}

}
